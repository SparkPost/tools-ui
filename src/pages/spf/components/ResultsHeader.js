import React from 'react';
import { CopyPopover } from 'components/popover/Popover';
import { ActionLink } from 'components/button/Button';
import classNames from 'classnames';

export default (props) => {
  const { domain, timestamp, authorized_netblocks, dns_lookups, refresh } = props;
  const timeClasses = classNames('text--muted', 'marginBottom--none', { 'h-hide': !timestamp });
  return (
    <div>
      <div className='panel panel--accent'>

        {/*  Top section */}
        <div className='panel__body'>
          <div className='float--right'>
            <CopyPopover><ActionLink>Share</ActionLink></CopyPopover>
            <ActionLink onClick={refresh}>Refresh</ActionLink>
            </div>
          <h1 className='marginBottom--none marginTop--xs'>{domain}</h1>
          <p className={timeClasses}>Tested on {timestamp}</p>
        </div>


        {/*  Info section */}
        <div className='panel__body'>
          <div className="flex">
            <div className="col-xs-3">
              <b>{authorized_netblocks}</b> Authorized Netblocks
            </div>
            <div className="col-xs-3">
              <b>{dns_lookups}</b> DNS Lookups
            </div>
          </div>
        </div>

      </div>
    </div>
  );
};
