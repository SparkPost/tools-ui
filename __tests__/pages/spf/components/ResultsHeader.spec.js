import React from 'react';
import renderer from 'react-test-renderer';
import ResultsHeader from 'pages/spf/components/ResultsHeader';

describe('ResultsHeader Snapshots', () => {
  let results
    , domain
    , refresh;

  beforeEach(() => {
    results = {
      timestamp: 'now',
      authorized_netblocks: 64,
      dns_lookups: 9
    };

    domain = 'itchyandscratchy.com';

    refresh = () => {}
  })

  test('baseline snapshot', () => {
    const panel = <ResultsHeader results={ results } domain={ domain } refresh={ refresh }></ResultsHeader>;
    expect(renderer.create(panel)).toMatchSnapshot();
  });

});
