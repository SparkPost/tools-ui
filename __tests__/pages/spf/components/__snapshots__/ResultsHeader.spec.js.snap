exports[`ResultsHeader Snapshots baseline snapshot 1`] = `
<div>
  <div
    className="panel panel--accent">
    <div
      className="panel__body">
      <div
        className="float--right">
        <span
          className="popover__group popover__hoverTrigger">
          <a
            className="actionLink"
            href="https://TEST.sparkpost.com/sign-up?return=http://localhost/&src=SP-Tools"
            title="Save Results">
            Save Results
          </a>
          <span
            className="popover popover--top popover--m">
            Create a free SparkPost account or login into your account to save results
          </span>
        </span>
        <span
          className="popover__group"
          onClick={[Function]}>
          <a
            className="actionLink"
            onClick={[Function]}
            style={Object {}}
            title="Share">
            Share
          </a>
          <div
            className="popover copyPopover popover--l popover--rightAligned popover--bottom">
            <h6
              className="text--left copyPopover__label">
              <i
                className="fa fa-check-circle " />
               Copied to Clipboard!
            </h6>
            <p
              className="copyPopover__copiedString">
              http://localhost/
            </p>
            <input
              className="copyPopover__hidden"
              readOnly={true}
              value="http://localhost/" />
          </div>
        </span>
        <a
          className="actionLink"
          onClick={[Function]}
          style={Object {}}
          title="Refresh">
          Refresh
        </a>
      </div>
      <h1
        className="marginBottom--none marginTop--xs">
        itchyandscratchy.com
      </h1>
      <p
        className="text--muted marginBottom--none h-hide">
        Tested on 
      </p>
    </div>
  </div>
</div>
`;
